[WIP]Energy weapon gameplay enhancement

Pockets-byte opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
this is an attempt to make energy weapons more reasonable for muni players by bringing them several more statistics that need to be managed, and one entire minigame definitely not borrowed from ds13
energy weapons degrade and lose effectiveness the more they're used, and muni needs to be on their toes to keep them functional and not exploding in your face. 

Credit goes to @Invictus282 for the heat sink sprite.
huge thanks to @Deltafire15 and codebus for coding help 

## Why It's Good For The Game

muni gameplay on the wisp real. 


there are three new stats to keep track of, weapon heat, weapon lens alignment, and weapon frequency alignment - should any of these reach their worst state, ::fun:: things happen, including weapon explosions, shutdowns, in-ship laser fire, and dangerous steam venting. 
## Testing Photographs and Procedure


<details>
<summary>Screenshots&Videos</summary>

![image](https://github.com/BeeStation/NSV13/assets/55299415/09f9faaf-a456-4d3d-9e69-db552b6ed4f0)
![image](https://github.com/BeeStation/NSV13/assets/55299415/5cca2037-3a29-40fe-b94d-ded7a95d8157)
![image](https://github.com/BeeStation/NSV13/assets/55299415/834c3b6a-9626-4947-808a-e6e2f0c5898a)
<img width="1539" height="925" alt="laserlaserlaser" src="https://github.com/user-attachments/assets/c20db605-bf98-421c-bb97-3da1fb74d6bb" />

</details>

## Changelog
:cl: Pockets-byte, Invictus282
add: Added several stats to keep track of to energy weapons
add: Added a minigame from ds13
balance:some rebalanced energy weapons to make up for required downtimes and more measured use
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Pockets-byte committed:
limb sprite addition test

hopefully will add another set of limbs to the augmentor once I finish  converting the sprites

Pockets-byte committed:
adds inital sprites for augments

mostly from paradise, I made the veymed ones from nsv sprites

Pockets-byte committed:
Added a number of augment skins from paradise/bay

Pockets-byte committed:
Add comments to augment icon lists

Pockets-byte committed:
Merge branch 'BeeStation:master' into patch-1

Pockets-byte committed:
moved augment sprites to NSV directory

Pockets-byte committed:
fixed the bad, replaced misplaced files, and fixed paths

Pockets-byte committed:
added a little more detail to the comment

Bokkiewokkie committed:
Update code/game/machinery/aug_manipulator.dm

Pockets-byte committed:
Merge branch 'BeeStation:master' into patch-1

Pockets-byte committed:
Merge branch 'BeeStation:master' into patch-1

Pockets-byte committed:
Merge branch 'BeeStation:master' into patch-1

Pockets-byte committed:
Merge branch 'BeeStation:master' into patch-1

Pockets-byte committed:
Merge branch 'BeeStation:master' into patch-1

Pockets-byte committed:
energy weapons rework?

gotta cherry pick this out

Pockets-byte committed:
blah minigame

Pockets-byte committed:
Update phaser.dm

Pockets-byte committed:
it works now

Pockets-byte committed:
A

Pockets-byte committed:
little bit of balancing stuff, some bugfixing

Pockets-byte committed:
AAAAAAAAAAAAAAAAA game no work

Pockets-byte committed:
tgui hell

Pockets-byte committed:
why

bluey89962 commented:
Question: would this also come with adding muni tech as a job to the whisp or would someone else (ie engineers/MAA/whoever the captain assigns) be responsible for this?

Pockets-byte commented:
> Question: would this also come with adding muni tech as a job to the whisp or would someone else (ie engineers/MAA/whoever the captain assigns) be responsible for this?

this most likely will come with muni jobs on the wisp, yes- these changes will make single-manning the wisp much harder- but probably not impossible

bluey89962 commented:
> this most likely will come with muni jobs on the wisp, yes- these changes will make single-manning the wisp much harder- but probably not impossible

I obviously don't know exactly how this system will work but in that case it might be good to restrict the amount of tecchies that we it can have because I expect we will many see cases where lots of people sign up for it since they see it as "the necessary job for NSV" (which it is on basically all ships) but no one goes for engineering, which is absolutely more essencial for the functioning of the whisp.

Of course it's all WIP and everything but I just thought I'd throw in my 2 cents considering the lack of engineers is a common issue for the map and it's a reason that people usually bring up as to why they dislike it.

Pockets-byte committed:
AAAAAAAAAAAAAA

Pockets-byte committed:
heat management stuff!

Pockets-byte committed:
magic cooling is really dead now

Pockets-byte committed:
AAAAAAAAAAAA

Pockets-byte committed:
removes slowprocess

Pockets-byte committed:
removes atmos cooling. beginning of game-magic cooling

Pockets-byte committed:
AAAAAAAAAAAAAaaaaaaaa

Pockets-byte committed:
Update Aetherwhisp2.dmm

Pockets-byte committed:
A

Pockets-byte committed:
blah

Pockets-byte committed:
cooling work

Pockets-byte committed:
icons for testing

Pockets-byte committed:
AAAAAAAAaaaaaaaaaaaaaaaaaaaa subspace cooler work

Pockets-byte committed:
"works"

Pockets-byte committed:
subspace heat management v1 partial completion

Pockets-byte committed:
no more punching subspace units

Pockets-byte committed:
hapazard balancing and prep for venting

Pockets-byte committed:
venting work, makes the BSA exempt from Energy mechanics

Pockets-byte committed:
venting things

Pockets-byte committed:
uncomments

Pockets-byte committed:
venting, and more balance stuff

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
fixes the damage done to the wisp and maps new energy weapon bits

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

DeltaFire15 committed:
I don't know what I'm doing

help

Co-Authored-By: Bokkiewokkie <43698041+Bokkiewokkie@users.noreply.github.com>
Co-Authored-By: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

test

Squashed commit of the following:

commit 945041d5e19de6337f92de8afeb7e0fc1ef38859
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Tue Jul 16 00:15:00 2024 +0000

    Automatic changelog compile [ci skip]

commit 9cb02c58b32436b3de2ec5535c6d957b600028df
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Sat Jul 13 22:04:04 2024 +0000

    Automatic changelog compile [ci skip]

commit 939793d5f9abffba6975b40cb7cac8a874a473ca
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Sat Jul 13 16:38:11 2024 -0500

    Automatic changelog generation for PR #2692 [ci skip]

commit ae25ec59924e10e062a902ccd86cb48450dd6d2b
Author: riquewright <123000630+riquewright@users.noreply.github.com>
Date:   Sat Jul 13 18:37:35 2024 -0300

    Fixes small typo in the NC reminder (#2692)

commit c649dead82dd5b03e8fd5c2a421f1a7a8cd499e7
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Sat Jul 13 16:37:02 2024 -0500

    Automatic changelog generation for PR #2689 [ci skip]

commit 8a2a79dd1460ac31df2cebd539b3cee6c2cc9aa6
Author: riquewright <123000630+riquewright@users.noreply.github.com>
Date:   Sat Jul 13 18:36:22 2024 -0300

    Re-adds the basic plasma cutter back into its respective node. (#2689)

commit dfeb98134dd90f6878a1a970a62be19b878414c4
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 12 19:03:52 2024 +0000

    Automatic changelog compile [ci skip]

commit 284e333be8e1b23cd083895ce17818772440453a
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 12 13:13:22 2024 -0500

    Automatic changelog generation for PR #2678 [ci skip]

commit 5408a3303d759a00ca02df13226a6f36c345302a
Author: DeltaFire <46569814+DeltaFire15@users.noreply.github.com>
Date:   Fri Jul 12 20:12:43 2024 +0200

    Only /overmaps are passed to after_enter() (#2678)

commit 1c03630d360e318adf63e3e80b7d3602f3c7a642
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 12 13:12:12 2024 -0500

    Automatic changelog generation for PR #2675 [ci skip]

commit ffed94265e7305f29353b6c4799b635a5da41a2d
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 12 13:11:57 2024 -0500

    Automatic changelog generation for PR #2676 [ci skip]

commit f3bbe830b0dac8379b7a90406c7e8d7410bb30ed
Author: Kenionatus <kenionatus@techie.com>
Date:   Fri Jul 12 20:11:33 2024 +0200

    Sort projectiles fx.dm better (#2675)

commit 08f874bef30144024d1e93a39b7794ff7dbcc4c8
Author: DeltaFire <46569814+DeltaFire15@users.noreply.github.com>
Date:   Fri Jul 12 20:10:55 2024 +0200

    Improves a little bit of cursed weapon code (#2676)

commit 7b2eb24c2d803e9d865a703a9ec3d831bb90abf4
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 12 18:05:04 2024 +0000

    Automatic changelog compile [ci skip]

commit c445ae1cb28cd0eaf97a919314f510721f2928a3
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 12 13:00:46 2024 -0500

    Automatic changelog generation for PR #2681 [ci skip]

commit c5326c6c0a46b8052284704593b54cb9c620c143
Author: Bokkiewokkie <43698041+Bokkiewokkie@users.noreply.github.com>
Date:   Fri Jul 12 20:00:09 2024 +0200

    Makes shields hits only spawn one visual (which is actually aligned with the ship) (#2681)

commit 580f275223757418c017765ae72b9ba1082d21fc
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 12 17:04:00 2024 +0000

    Automatic changelog compile [ci skip]

commit ac2fc1caff7ee082a18089b6b056fb07c5245bb5
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 12 11:54:55 2024 -0500

    Automatic changelog generation for PR #2671 [ci skip]

commit 9ea99c5527bf157125bb09617dbb5ffd5730ce40
Author: ᛈᛟᚲᚲᛖᛏᛋ <55299415+Pockets-byte@users.noreply.github.com>
Date:   Fri Jul 12 10:54:19 2024 -0600

    adds a playable solgov fighter, and a fighter lasercannon  (#2671)

commit c6497319161b7253bf87e5811fd2ce089abb2dbb
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 12 11:52:43 2024 -0500

    Automatic changelog generation for PR #2669 [ci skip]

commit 6ced45c091ad0fb516d9bbacaa220c439e3f9eff
Author: ᛈᛟᚲᚲᛖᛏᛋ <55299415+Pockets-byte@users.noreply.github.com>
Date:   Fri Jul 12 10:52:00 2024 -0600

    Adds a solgov variant of the squad vendor (#2669)

commit ee6f6371a3843dd154919decd32d52c40af0465b
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 5 22:04:41 2024 +0000

    Automatic changelog compile [ci skip]

commit e5ceeaff7d0e4d5e61b339e5487963ad798c301e
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 5 16:16:16 2024 -0500

    Automatic changelog generation for PR #2670 [ci skip]

commit 6337c914aa6b10197cd192bc12be7d386e726c1f
Author: ᛈᛟᚲᚲᛖᛏᛋ <55299415+Pockets-byte@users.noreply.github.com>
Date:   Fri Jul 5 15:15:36 2024 -0600

    Adds relayed laser projectiles (#2670)

    Co-authored-by: Bokkiewokkie <43698041+Bokkiewokkie@users.noreply.github.com>

commit e7abbccbe50c9aa59beb3f38300ca7a224a0ae84
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jul 5 00:14:35 2024 +0000

    Automatic changelog compile [ci skip]

commit 03f48076178c8a7bc75ca125a03a046df26107f4
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 28 19:04:11 2024 +0000

    Automatic changelog compile [ci skip]

commit 3d8738541f5206dd95cffa09ee932be9105f8f81
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 28 13:51:16 2024 -0500

    Automatic changelog generation for PR #2663 [ci skip]

commit 2b48323980b2d76bf271ec25ca4e19e83e73fb92
Author: DeltaFire <46569814+DeltaFire15@users.noreply.github.com>
Date:   Fri Jun 28 20:50:39 2024 +0200

    Removes integrity failure value from all overmap entities (#2663)

commit e1004ae5cc211bf62207105966648b9420e01d24
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Sun Jun 23 00:15:57 2024 +0000

    Automatic changelog compile [ci skip]

commit 0ee1b108635168f7386a5c3b6b56615bab0df11a
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 21 22:04:33 2024 +0000

    Automatic changelog compile [ci skip]

commit 386288e99d39209c8a895ba323aee9e366050955
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 21 16:50:06 2024 -0500

    Automatic changelog generation for PR #2668 [ci skip]

commit 879aea1ac21805aa605c1cbb541a91e4af64db36
Author: Bokkiewokkie <43698041+Bokkiewokkie@users.noreply.github.com>
Date:   Fri Jun 21 23:49:26 2024 +0200

    Updates the armour pumps to account for lag properly (#2668)

commit f844e33f755c58d435731c44391e836808cc3f3e
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 21 16:48:40 2024 -0500

    Automatic changelog generation for PR #2661 [ci skip]

commit 501304dfbb4a74a33579fd48715e833aaf48b804
Author: SerynEngi <75588150+SerynEngi@users.noreply.github.com>
Date:   Fri Jun 21 14:48:04 2024 -0700

    Atlas Armor Alterations (#2661)

commit aaf1109dcd383d170b457ba38d4b9fb67b9793b3
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 21 16:46:25 2024 -0500

    Automatic changelog generation for PR #2653 [ci skip]

commit cf8d193f6f8357f4dc9657da1516071b482a73e6
Author: Bokkiewokkie <43698041+Bokkiewokkie@users.noreply.github.com>
Date:   Fri Jun 21 23:45:49 2024 +0200

    Properly registers small craft in overmap ships and fixes sscrit explosions (#2653)

commit 56ee1053e947800c6b7fb08b7787c7e094477fac
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 21 00:14:15 2024 +0000

    Automatic changelog compile [ci skip]

commit 6d0981eef4d43a44b7e7a3c93e2c018e9f32487e
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 14 21:03:53 2024 +0000

    Automatic changelog compile [ci skip]

commit 48d538ae333459c3711b7764a4ed643b26a59619
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 14 15:31:57 2024 -0500

    Automatic changelog generation for PR #2665 [ci skip]

commit 2b0d18cdd6e8ae89080729966a0c4fc7d8ae9372
Author: SerynEngi <75588150+SerynEngi@users.noreply.github.com>
Date:   Fri Jun 14 13:31:22 2024 -0700

    Even more Atlas Mapping Fixes (#2665)

commit b31ef9d66467d132d7d42bc11cfcce3a59f077d7
Author: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Date:   Fri Jun 14 22:23:33 2024 +0200

    Bump braces from 3.0.2 to 3.0.3 in /tgui (#2682)

    Signed-off-by: dependabot[bot] <support@github.com>
    Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

commit cfe01083974516b7a02b92ddf42c56daed4a71ad
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 14 15:21:55 2024 -0500

    Automatic changelog generation for PR #2659 [ci skip]

commit 0b47f836d66d03d7ae935be6b5323ae8c2d1f129
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 14 15:21:48 2024 -0500

    Automatic changelog generation for PR #2660 [ci skip]

commit a6a4a7c87947d2fe48c60d5415019c37306202f2
Author: IndusRobot <whiz_kid10@yahoo.com>
Date:   Fri Jun 14 14:21:17 2024 -0600

    Fixes Shells are not appropriately handling the presence of powder bags. (#2659)

commit 01caa93c4add09a579143b8b9a04818d8a4baa4b
Author: Bokkiewokkie <43698041+Bokkiewokkie@users.noreply.github.com>
Date:   Fri Jun 14 22:21:05 2024 +0200

    Adds the ability to (de)construct pool ladders (#2660)

commit 06811529ce73ec6329d7d2b8b5eba98c2462298e
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 14 15:20:01 2024 -0500

    Automatic changelog generation for PR #2657 [ci skip]

commit 9657baf057e7bd366790be55ced43986df01ec75
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 14 15:19:26 2024 -0500

    Automatic changelog generation for PR #2658 [ci skip]

commit 57cd724dadaf93c1503a62e2452f7a4d624be521
Author: IndusRobot <whiz_kid10@yahoo.com>
Date:   Fri Jun 14 14:19:21 2024 -0600

    Fixes borgs duplicating ammunition with the munition clamp upgrade (#2657)

commit 37579a6c87cfa8c3a9c6651781736aef10af3a6d
Author: Bokkiewokkie <43698041+Bokkiewokkie@users.noreply.github.com>
Date:   Fri Jun 14 22:18:48 2024 +0200

    Fixes some Engineering design issues and Hangar airlocks on Gladius (#2658)

commit 0c4a4fff2218b45013cc100155cd07aea1a7be7d
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 14 19:04:57 2024 +0000

    Automatic changelog compile [ci skip]

commit cf293aeb7bf31fc5f85ed57693997cb734e7e455
Author: BlueHNT <79374236+BlueHNT@users.noreply.github.com>
Date:   Fri Jun 14 20:29:29 2024 +0200

    Updates forgotten armor values and syndicate armor stacking during boarding (#2638)

commit 4a8b53ae5c2c708220dd08af2c986f6f047b1e71
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 14 13:23:36 2024 -0500

    Automatic changelog generation for PR #2646 [ci skip]

commit d067eee4b2138d22143ee7843bcf999b8dc6e610
Author: DeltaFire <46569814+DeltaFire15@users.noreply.github.com>
Date:   Fri Jun 14 20:22:52 2024 +0200

    Adds digitigrade robotic legs (#2646)

commit afe5367c5af66d3ff8fc0684802e45a090da8c66
Author: ss13-beebot <56381746+ss13-beebot@users.noreply.github.com>
Date:   Fri Jun 14 16:04:03 2024 +0000

    Automatic changelog compile [ci skip]

Revert "Squashed commit of the following:"

This reverts commit 57ff137d55fe0c481dd1032c5aed36f04c025f22.

Revert "test"

This reverts commit 0ee85848796eeb7924f936696a315360f6010404.

Revert "I don't know what I'm doing"

This reverts commit 197c80d6c8c9be68859e1745713e8958d4faadd1.

Pockets-byte committed:
Research and fixing comments

the gameplay loop theoretically works now

Pockets-byte committed:
please let this fix the merge conflict

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
fixes everything

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
Update job_changes.dm temporarily

Pockets-byte committed:
Update job_changes.dm

Bokkiewokkie commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -159,3 +255,139 @@
+	var/turf/detonation_turf = get_turf(src)
+	if(heat >= (3*(max_heat/4)))
+		freq -= rand(1,4)
+	if(alignment <= 75)
+		if(prob(50))
+			do_sparks(4, FALSE, src)
+			freq -= rand(1,10)
+	if(alignment <= 50)
+		if(prob(45))
+			do_sparks(4, FALSE, src)
+			freq -= rand(1,10)
+			playsound(src, malfunction_sound, 100, 1)
+		if(prob(5))
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+			explosion(detonation_turf, 0, 0, 2, 3, flame_range = 2)
+	if(alignment <= 25)
+		if(prob(25))
+			do_sparks(4, FALSE, src)
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+		if(prob(25))
+			playsound(src, malfunction_sound, 100, 1)
+			freq -= rand(1,10)
+			explosion(detonation_turf, 0, 0, 3, 4, flame_range = 3)
+		if(prob(50))
+			var/list/shootat_turf = RANGE_TURFS(5,detonation_turf) - RANGE_TURFS(4, detonation_turf)
+			var/obj/item/projectile/beam/laser/P = new(detonation_turf)
+			//Shooting Code:
+			P.range = 6
+			P.preparePixelProjectile(pick(shootat_turf), detonation_turf)
+			P.fire()
+			freq -= rand(1,10)
+	alignment = max(alignment-(rand(0, 4)),0)
```
```suggestion
	switch(alignment)
		if(51 to 75)
			if(prob(50))
				do_sparks(4, FALSE, src)
				freq -= rand(1,10)
		if(26 to 50)
			var/roll = roll(1,20)
			switch(roll)
				if(1 to 9)
					do_sparks(4, FALSE, src)
					freq -= rand(1,10)
					playsound(src, malfunction_sound, 100, 1)
				if(10)
					playsound(src, malfunction_sound, 100, 1)
					freq -= rand(1,10)
					explosion(detonation_turf, 0, 0, 2, 3, flame_range = 2)
		if(0 to 25)
			var/roll2 = roll(1,4)
			switch(roll2)
				if(1)
					do_sparks(4, FALSE, src)
					playsound(src, malfunction_sound, 100, 1)
					freq -= rand(1,10)
				if(2)
					playsound(src, malfunction_sound, 100, 1)
					freq -= rand(1,10)
					explosion(detonation_turf, 0, 0, 3, 4, flame_range = 3)
				if(3,4)
					var/list/shootat_turf = RANGE_TURFS(5,detonation_turf) - RANGE_TURFS(4, detonation_turf)
					var/obj/item/projectile/beam/laser/P = new(detonation_turf)
					//Shooting Code:
					P.range = 6
					P.preparePixelProjectile(pick(shootat_turf), detonation_turf)
					P.fire()
					freq -= rand(1,10)
	alignment = max(alignment-(rand(0, 4)),0)
```
The way you have written this originally, all of these conditions will fire at the same time if alignment is < 25, so use a switch statement instead. I also added switch statements to your switch statements, because I figured you didn't want all of your probs which suspiciously added up to 100% and did a bunch of the same stuff to proc at the same time either.

Bokkiewokkie commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -159,3 +255,139 @@
+		to_chat(user, "<span class='warning'>You inputted [dowhat] into the command sequence.</span>")
+		playsound(src, 'sound/machines/sm/supermatter3.ogg', 20, 1)
+		if(combocount <= 4)
+			addtimer(CALLBACK(src, TYPE_PROC_REF(/atom,screwdriver_act), user),2)   // *scream  addtimer(CALLBACK(object|null, GLOBAL_PROC_REF(type/path|procstring), arg1, arg2, ... argn), time, timertype)
```
This looks terrible, why not use a for loop to continuously let people insert new things until they've added 5? (Or if they press "cancel" somewhere)

Pockets-byte replied:
Some of this is repaired ds13 code. I don't exactly know how to recreate it or modify it. I think it was originally supposed to be timed

Bokkiewokkie commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -159,3 +255,139 @@
+	. = ..()
+	if(maint_state == MSTATE_UNBOLTED)
+		.=TRUE
+		var/sound/thesound = pick('nsv13/sound/effects/computer/beep.ogg','nsv13/sound/effects/computer/beep2.ogg','nsv13/sound/effects/computer/beep3.ogg','nsv13/sound/effects/computer/beep4.ogg','nsv13/sound/effects/computer/beep5.ogg','nsv13/sound/effects/computer/beep6.ogg','nsv13/sound/effects/computer/beep7.ogg','nsv13/sound/effects/computer/beep8.ogg','nsv13/sound/effects/computer/beep9.ogg','ns ...
```
Note for a different PR: compile this into like a global list or just use a different system.
(nothing you have to worry about in your PR, you can add a comment about it if you want.)

Pockets-byte replied:
this was originally a list from ds13, but the sounds were in a completely different place in nsv, and I figured nobody else would use this, so there's no reason to make a list- then again it makes it more readable and stuff

Bokkiewokkie replied:
iirc we use these for console interactions as well, and you using it here proves that there may be more future use cases, so a list would be way easier

Bokkiewokkie commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -159,3 +255,139 @@
+				combo = null
+
+/obj/machinery/ship_weapon/energy/multitool_act(mob/living/user, obj/item/multitool/I)
+	if(maint_state == MSTATE_CLOSED)
+		if (istype(I))
+			to_chat(user, "<span class='notice'>You log [src] in the multitool's buffer.</span>")
+			I.buffer = src
+			return TRUE
+	if(maint_state == MSTATE_UNSCREWED)
+		to_chat(user, "<span class='notice'>You must unbolt the protective casing before aligning the lenses!</span>")
+	if(maint_state == MSTATE_UNBOLTED)
+		. = TRUE
+		to_chat(user, "<span class='notice'>You being aligning the lenses.</span>")
+		while(alignment < 100)
+			if(!do_after(user, 5, target = src))
+				return
+			alignment += rand(1,2)
+			if(alignment >= 100)
+				alignment = 100
+				break
+
```
```suggestion
	switch(maint_state)
		if(MSTATE_CLOSED)
			if(istype(I))
				to_chat(user, "<span class='notice'>You log [src] in the multitool's buffer.</span>")
				I.buffer = src
				return TRUE
		if(MSTATE_UNSCREWED)
			to_chat(user, "<span class='notice'>You must <I>unbolt</I> the protective casing before aligning the lenses!</span>")
			return FALSE
		if(MSTATE_UNBOLTED)
			to_chat(user, "<span class='notice'>You being aligning the lenses.</span>")
			while(alignment < 100)
				if(!do_after(user, 5, target = src))
					return ..()
				alignment += rand(1,2)
				if(alignment >= 100)
					alignment = 100
					to_chat(user, "<span class='notice'>You finish aligning the lenses.</span>")
					return TRUE
	return ..()
```	
Since it uses states, just a switch is easier. I also cleaned up returns and ..() a bit to make more sense.

Pockets-byte replied:
what does ..() do?

Bokkiewokkie commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm:
```diff
@@ -0,0 +1,107 @@
+	desc = "A cooling unit that stores the massive amounts of heat energy weapons generate in subspace."
+	icon_state = "smes"
+	circuit = /obj/item/circuitboard/machine/cooling/storage
+
```
I'm going to be honest, I don't like how most of this file and code is written, would you like me to rewrite it or do you prefer that I just give you tips?
If you do not have any sprites yet, I would be happy to make some. Even if you don't, it is usually appreciated to have *some* distinguishing features between your own machines and already existing ones. Even if it's only a red and blue stripe, just those things can be enough for telling apart two components.

Pockets-byte replied:
I have plans to put together sprites, but I haven't taken any steps to do so, if you want to add some sprites, either placeholder or fully completed, that wouldn't be bad at all

Bokkiewokkie commented on nsv13/code/modules/research/designs/ship_weapon_designs.dm:
```diff
@@ -200,3 +200,25 @@
+	id = "subspace_cooler"
+	build_type = PROTOLATHE|IMPRINTER
+	materials = list(/datum/material/glass = 2000, /datum/material/copper = 2000, /datum/material/gold = 5000)
+	build_path = /obj/item/circuitboard/machine/cooling
```
```suggestion
	build_path = /obj/item/circuitboard/machine/cooling/cooler
```
Otherwise this does nothing

Pockets-byte replied:
oops

Pockets-byte replied:
wait there's no cooler board, the base prefab for the board is the cooler board. this breaks

Bokkiewokkie commented on nsv13/code/modules/research/designs/ship_weapon_designs.dm:
```diff
@@ -200,3 +200,25 @@
+
+//subspace cooling stuff
+/datum/design/board/subspace_cooler
+	name = "Machine Design (Experemental Subspace Cooling Unit)"
```
```suggestion
	name = "Machine Design (Experimental Subspace Cooling Unit)"
```

Bokkiewokkie commented on nsv13/code/modules/research/designs/ship_weapon_designs.dm:
```diff
@@ -200,3 +200,25 @@
+	departmental_flags = DEPARTMENTAL_FLAG_MUNITIONS
+
+/datum/design/board/subspace_storage
+	name = "Machine Design (Experemental Subspace Storage Unit)"
```
```suggestion
	name = "Machine Design (Experimental Subspace Storage Unit)"
```

Bokkiewokkie commented on nsv13/code/modules/research/techweb/all_nsv_nodes.dm:
```diff
@@ -261,3 +261,15 @@
 	export_price = 5000
+
+/// Munitions Upgrade for Energy Weapons
+/datum/techweb_node/Weapon_Cooling
+	id = "Weapon_Cooling"
```
```suggestion
/datum/techweb_node/weapon_cooling
	id = "weapon_cooling"
```
Please try to avoid capitalizing typepaths, and also id names in this case.

Bokkiewokkie commented on nsv13_unit_test_focus_file.dm:
```diff
@@ -0,0 +1,42 @@
+TEST_FOCUS(/datum/unit_test/subsystem_init)
+TEST_FOCUS(/datum/unit_test/subsystem_metric_sanity)
+TEST_FOCUS(/datum/unit_test/tgui_create_message)
+TEST_FOCUS(/datum/unit_test/timer_sanity)
```
This file should be deleted

Bokkiewokkie commented on tgui/packages/tgui/interfaces/EnergyWeapons.js:
```diff
@@ -38,6 +42,20 @@ export const EnergyWeapons = (props, context) => {
           <br />
+          Heat:
+          <ProgressBar
+            value={(heat / maxheat * 100) * 0.01}
```
```suggestion
            value={(heat / maxheat)}
```
`100*0.01 = 1`

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -14,13 +21,33 @@
+	heat_per_shot = 100
+	var/max_freq = 100
+	var/combo_target = "omega" //Randomized sequence for the recalibration minigame.
+	var/list/letters = list("delta,", "omega,", "phi,")
```
it was like that on ds13 too, and the way it's set up, it looks weird when examined. it also looks weird with the commas, but less so- it's been quite a while. but I don't know a lot about lists- it's displayed on the weapon it pertains to

Pockets-byte replied:
it's displayed in the next file, `/obj/machinery/ship_weapon/energy/examine(mob/user)
	. = ..()
	if(in_range(user, src) || isobserver(user))
		. += "<span class='notice'>The heatsink display reads <b>[(heat)]</b> out of <b>[(max_heat)]</b>.</span>"
		if(maint_state != MSTATE_CLOSED)
			. +=  "<span class='warning'>[src]'s realignment sequence is: [combo_target].</span>"`

Pockets-byte replied:
I tinkered with trying to improve it, but I couldn't make it work without sticking to the original design

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -159,3 +255,139 @@
+	var/H = heat-cooling_amount*heat_rate
+	if(heat > 0)
+		heat = max((H),0)
+	if((weapon_state == STATE_OVERLOAD) & (heat <= (max_heat/50)))
+		weapon_state = STATE_NOTHING
+	if(weapon_state == STATE_OVERLOAD)
+		return
+	if(weapon_state == STATE_VENTING)
+		if(heat <= max_heat-ventnumber)
+			weapon_state = STATE_NOTHING
+			return
+		heat = max(((H*1.2)-H),0)
+		return
```
that... was the intention. I have no idea what I was thinking
I don't know how to use switch()- I probably should learn

Pockets-byte committed:
Delete nsv13_unit_test_focus_file.dm

Pockets-byte committed:
Apply suggestions from code review

Co-authored-by: Bokkiewokkie <43698041+Bokkiewokkie@users.noreply.github.com>

Pockets-byte committed:
Delete job_changes.dm

plushie sailing into distance.gif

Pockets-byte committed:
mistake

Pockets-byte committed:
Revert "Delete job_changes.dm"

This reverts commit 733e882f12a5b4cb7b03cef773aa228b42ec8e41.

Pockets-byte committed:
MINOR SPELLING MISTAKE

Death

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
new sprite

Pockets-byte committed:
applies sprites

DeltaFire15 commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm:
```diff
@@ -0,0 +1,107 @@
+/obj/machinery/cooling/cooler/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/ship_weapon/energy/E = locate(/obj/machinery/ship_weapon/energy) in orange(1, src) //I have no idea what I'm doing and this causes errors so
+	E.coolers |= src
+	parent = E
```
Unhandled references. Null parent on own `Destroy` and also remove from lists / clear parent relation if the parent object is being `Destroy()`ed.

Pockets-byte replied:
It does that in the below lines doesn't it?

DeltaFire15 replied:
> Should I also put parent = null below parent.coolers = null?

`parent = null` should come after you do things related to the parent in the `Destroy`, yes,

However you should not null `parent.coolers`here, are parent still exists, that part is correct; What you are missing is handling for if parent is being deleted.

DeltaFire15 commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm:
```diff
@@ -0,0 +1,107 @@
+/obj/machinery/cooling/storage/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/ship_weapon/energy/E = locate(/obj/machinery/ship_weapon/energy) in orange(1, src)
+	E.storages |= src
+	parent = E
```
As with above, reference handling.

DeltaFire15 commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm:
```diff
@@ -0,0 +1,107 @@
+		. += "The power is on"
+	if(parent)
+		. += "The thermal subspace transcever is linked"
+	else
+		. += "it's completely inactive"
```
Can be on and also inactive which will print both messages that are somewhat contradictory. Missed an `else if` above?

Pockets-byte replied:
These are mostly debug messages, so printing both is still helpful even if confusing

Pockets-byte replied:
despite this, should I add additional handling or re-phrase it?

DeltaFire15 replied:
Given it is currently present as the PR is testmerged, it would be useful if examine information was precise.

Pockets-byte replied:
I'll see if I can add additional handling

Pockets-byte replied:
this is basically all the existing states FOR the cooling machines, so I'm not sure that they can actually have additional handling?

DeltaFire15 commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm:
```diff
@@ -0,0 +1,107 @@
+//		add_overlay("smes-op0")
+
+
+/obj/machinery/cooling/multitool_act(mob/living/user, obj/item/multitool/I)
```
This proc does not check whether this machine is already linked, which allows you to link one cooler to several weapons.

Pockets-byte replied:
Aw shoot. I've got to fix that. I completely didn't think of that

Pockets-byte replied:
So maybe do parent.coolers = null and then parent = null before handling adding itself to another weapon?

Pockets-byte replied:
Or alternatively, if parent.coolers = src then return false?

DeltaFire15 replied:
>So maybe do parent.coolers = null and then parent = null before handling adding itself to another weapon?

Yesn't. You would have to remove it from the list, however *nulling* `parent.coolers` would drop the *entire* list (all coolers, also the fact that variable was a list). See list operators for removal of a specific element (`- / -=` removal / `+ / +=` add)

Pockets-byte replied:
yeah that was my brain being stupid, I saw that just now- I want the same handling as in destroy, parent.coolers =- src or however it goes

DeltaFire15 commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm:
```diff
@@ -0,0 +1,107 @@
+			.=TRUE
+			parent.storages |= src
+			to_chat(user, "<font color = #666633>-% Successfully linked [P.buffer] with [src] %-</font color>")
+		if(length(parent.storages)+length(parent.coolers) >= 10)
+			var/E = pick(parent.storages+parent.coolers)
+			explosion(get_turf(E), 0, 1, 3, 5, flame_range = 4)
+			return
```
This fact (maximum linkage count) should really be told to the user, potentially within the energy weapon `examine` chain

Pockets-byte replied:
That's probably a good idea.   I don't want to be too cruel

Pockets-byte replied:
it now lists actual percentage connection capacity used

DeltaFire15 reviewed (changes requested):
Preliminary.

Also maintenance on energy weapons should have a lockout var to prevent several users from interacting with one at the same time (or worse, one user several times at once).

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/_ship_weapon.dm:
```diff
@@ -75,9 +75,13 @@
 
 	// These variables only pertain to energy weapons, but need to be checked later in /proc/fire
 	var/charge = 0
+	var/heat = 0
 	var/charge_rate = 0 //How quickly do we charge?
 	var/charge_per_shot = 0 //How much power per shot do we have to use?
-
+	var/heat_per_shot = 0 //how much heat do we make per shot
+	var/heat_rate = 0 // how fast do we discharge heat
+	var/max_heat = 0 //how much heat before ::fun:: happens
+	var/overloaded = 0 //have we cooked ourself
```
Should I move them all?

DeltaFire15 replied:
Should be good, yeah, assuming I haven't missed any usage in the base procs.

Pockets-byte commented on nsv13/code/game/objects/items/nsv_circuitboards.dm:
```diff
@@ -477,3 +477,33 @@
+	req_components = list(
+		/obj/item/stack/cable_coil = 30,
+		/obj/item/stock_parts/matter_bin = 4,
+		/obj/item/stack/sheet/mineral/copper = 30,
```
I do think it makes sense for a heat related machine, and unless you don't want functional weapons, you can't scrap them, then again. 
I don't want them to be super cheap though, should I make the boards more expensive in that case?

DeltaFire15 replied:
You could use custom components which are expensive to print, for example; This makes them somewhat harder to efficiently cannibalize (and add a layer of ambiguosity), over just ejecting & slamming in a stack of 30 copper.

Pockets-byte replied:
That's true. Probably a much cleverer idea than just slamming in a minecart's worth of raw metal
It will need sprites tho

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/bsa.dm:
```diff
@@ -227,3 +227,10 @@
 	explosion(get_turf(target), 6, 8, 9, 12, ignorecap = TRUE, flame_range = 6) //I have to keep myself from letting it just truncate ships because thats a bit annoying to fix for the receiving side, even if accurate to appearance.
+
+
+/obj/machinery/ship_weapon/energy/beam/bsa/after_fire()  //NT gets to cheat and have no degadation or heating on their weapon because muni has enough to do and the BSA doesn't need buffs/nerfs
+	. = ..()
+	heat = 0
+	alignment = 100
+	freq = 100
```
I'd have to add a check for that in every part of the system though right? Would that be better

DeltaFire15 replied:
It's the classic "address now vs. address later" issue.
For now it's probably fine with a single energy weapon that works this way, but if more appear, it may get a bit less convenient.
(Besides the fact you can do all the related maintenance steps on the BSA with this most likely despite them having no effect, and even get the examine talking about these things.)

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -159,3 +255,148 @@
+				return TRUE
+		if(MSTATE_UNSCREWED)
+			to_chat(user, "<span class='notice'>You must <I>unbolt</I> the protective casing before aligning the lenses!</span>")
+			return FALSE
```
Oh darn. Thanks, I'll get that fixed

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -159,3 +255,148 @@
+	freq = 0
+	say("WARNING! Critical heat density, emergency venting and shutdown initiated!")
+	atmos_spawn_air("water_vapor=200;TEMP=1000")
+	heat = max_heat
```
Hm. I need to make sure this can't trigger if overloaded is already there, so I should put that at the start

Pockets-byte replied:
Wait actually it doesn't trigger overload when overloaded because the process loop stops before it triggers

Pockets-byte replied:
Anyway it'll become pretty dangerous if they keep overloading it(at least that's the intent). It's safe with a couple overloads, but should start boiling them if they keep doing it

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -49,6 +95,16 @@
+/obj/machinery/ship_weapon/energy/proc/vent()
+	if(heat > max_heat/4)
+		weapon_state = STATE_VENTING
+		ventnumber = max_heat - (3*(max_heat/4))
```
I'm not sure even sure why this is like this honestly. There probably was a reason in the past but...

Pockets-byte committed:
Update phaser.dm

Co-authored-by: DeltaFire <46569814+DeltaFire15@users.noreply.github.com>

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -107,6 +169,20 @@
+			if(M.stat != DEAD && M.get_eye_protection() < 2) //checks for eye protec
+				M.flash_act(10)
+				to_chat(M, "<span class='warning'>You have a second to watch the casing of the gun glow a dull red before it erupts in a blinding flash as it self-destructs</span>")   // stealing this from the plasmagun as well
+		explosion(get_turf(src), 0, 1, 3, 5, flame_range = 4)
+		overload()
 	else
 		return TRUE
```
What actually does this do?

DeltaFire15 replied:
base return value is null, which, while similar to FALSE, is not *exactly* FALSE. If a proc always returns TRUE / FALSE, it should generally do that for all cases for consistency's sake.
Consequently, with that present, since any `if` that is fulfilled returns without reaching the end, the else is not neccessary.

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -14,13 +21,33 @@
 	var/power_modifier_cap = 3 //Which means that your guns are spitting bursts that do 60 damage.
 	var/energy_weapon_type = /datum/ship_weapon/burst_phaser
 	var/static_charge = FALSE //Controls whether power and energy cost scale with power modifier. True = no scaling
+	var/alignment = 100 //stolen from railguns and the plasma gun
```
How do I actually do that? I tried to replicate the style of the other comments

DeltaFire15 replied:
The second half of my comment states what sort of syntax is (most commonly) used to autodoc.

Pockets-byte commented on nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm:
```diff
@@ -159,3 +255,148 @@
+				weapon_state = STATE_NOTHING
+			else
+				return
+		if(STATE_VENTING)
+			if(heat <= max_heat-ventnumber)
+				weapon_state = STATE_NOTHING
+				return
+			heat = max(H*0.2,0)
+			return
```
This doesn't look quite right yeah, but I don't remember

Pockets-byte replied:
Should* be fixed now. 
Fixing things on mobile is a little squick, but this is a big one

Pockets-byte commented:
> Preliminary.
> 
> 
> 
> Also maintenance on energy weapons should have a lockout var to prevent several users from interacting with one at the same time (or worse, one user several times at once).

How do I actually do that? Just like a var that's set while someone's doing something? I should take a look at how the other weapons do it

Pockets-byte committed:
Update phaser.dm

Co-authored-by: DeltaFire <46569814+DeltaFire15@users.noreply.github.com>

Pockets-byte committed:
Fixes venting, maybe?

Sus

DeltaFire15 commented:
> How do I actually do that? Just like a var that's set while someone's doing something? I should take a look at how the other weapons do it

Variable on weapon that is set when someone starts an action on it and unset when it is either completed or failed. (`being_maintained` or whatever).

Pockets-byte committed:
fixes some of the issues, moves energy weapon vars, etc

more examine states

Pockets-byte committed:
storage sprite

Pockets-byte committed:
null reference exception!

Pockets-byte committed:
Update nsv13/code/modules/munitions/ship_weapons/energy_weapons/phaser.dm

Co-authored-by: DeltaFire <46569814+DeltaFire15@users.noreply.github.com>

Pockets-byte committed:
small stuff

Pockets-byte committed:
no more multi-linking

Pockets-byte committed:
Update subspacecooling.dm

Pockets-byte committed:
"quick" fix

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Bokkiewokkie reviewed (changes requested):
Mapping wise, I would rather you either expand the rooms to actually have the associated machines next to their respective lasers, or move things around so it is generally consistent where they all are (specifically noticing the heat sinks being all over the place)

Bokkiewokkie commented:
UnTMing for now, I hope the testing ingame so far has been helpful

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
funny stock item for de-looting, and variables to use later

Pockets-byte committed:
comment

Pockets-byte committed:
Merge branch 'energy-weapon-gameplay' of https://github.com/Pockets-byte/NSV13 into energy-weapon-gameplay

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
Variable name was taken

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
Merge remote-tracking branch 'upstream/master' into energy-weapon-gameplay

Pockets-byte committed:
fixed wisp mapping. gun maint lockout. heatsink sprite, partially animated

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
Merge remote-tracking branch 'upstream/master' into energy-weapon-gameplay

Pockets-byte committed:
returns the variables....

Pockets-byte committed:
slight messup

This reverts commit cf1fce4d65f765acceeccac8ed6c5638f83af1f4.

Revert "returns the variables...."

This reverts commit 6bedf044d299c5c7907a1b0e8fba03ba93c15b30.

Revert "Merge remote-tracking branch 'upstream/master' into energy-weapon-gameplay"

This reverts commit 0340693b4a3175e1ec4301257bc93697bdadd010, reversing
changes made to b59cbfc7467f1886ae71001e7a69b11f0aac46fa.

Pockets-byte committed:
AAAAAAAAAAAAA

Pockets-byte committed:
help.

Pockets-byte committed:
might've fixed it

Pockets-byte committed:
puts the maintainable var into prefabs and explodes the hack on the BSA

Pockets-byte committed:
I don't know how to use VSCODE

Pockets-byte committed:
LIGHT!

Pockets-byte committed:
touches the whisp gunnery room. fixes cooler decon. blurb about maintainablity

Pockets-byte committed:
inital minigame efforts
I have a headache

Pockets-byte committed:
ds13 minigame works now

Pockets-byte committed:
range check

Pockets-byte committed:
autodoc commenting. minigame continues on it's own

Pockets-byte committed:
arbitrary energy weapon buffs to compensate for having lower cadence of fire

Pockets-byte committed:
laser pdcs no longer require maintinece. slight buffs to all phaser weapons

Pockets-byte committed:
buffs burst phaser heat per shot because they're kinda weak

Pockets-byte committed:
test balance stuff

Pockets-byte committed:
slightly buffs ams because of the requirement to maintain it now

Pockets-byte committed:
I don't know why this was changed.

Pockets-byte committed:
overload anims. ion cannon sprites.

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

DuskLight9978 commented:
This is currently test merged, someone with stronger fists than me please smack that label in

Pockets-byte committed:
commens out overlays

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
slight balancing pass, bug fixes

Pockets-byte committed:
makes burst phasers stronger, and gives them a heat based boost in damage as they get hotter up to 20% stronger as well as a litterally doubled power cap

Pockets-byte committed:
makes deconstructing energy weapons to avoid having to maintain them a bad idea

Pockets-byte committed:
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

Pockets-byte committed:
I AM SO MAD

Pockets-byte committed:
removes map changes for multi-targeting testing

Pockets-byte committed:
bla

Pockets-byte committed:
gimballed pulse phasers

Pockets-byte committed:
fixes mistake in after fire()

Pockets-byte committed:
maint_wear() added

Pockets-byte committed:
overlay no worky

Pockets-byte committed:
masking stuff

Pockets-byte committed:
testing testing testing testing

Pockets-byte committed:
icon state

Pockets-byte committed:
blah

Pockets-byte committed:
AAAAA

Pockets-byte committed:
ack.

Pockets-byte committed:
filterfilterfilterfilterfilterfilterfilterfilterfilterfilterfilterfilterfilterfilter

Pockets-byte committed:
temperature gauge on energy weapons

huge thanks to deltafire and artur from codebus

Pockets-byte committed:
null handling

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
I might've forgotten to commit

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay

Pockets-byte committed:
vnc energy-ified

this is going to give me no end of headaches later. but that's a problem for later me

Pockets-byte committed:
revamped overheat gauge

now operates on a dual bar system that has a bar for average heat, and a bar for hottest weapon. fixes inconsistent cooling updates

Pockets-byte committed:
makes HUD not update when there's nothing to update

Pockets-byte committed:
Merge branch 'BeeStation:master' into energy-weapon-gameplay
